File: LeetCode/go/leetcode/isHappy.go
package leetcode

func getTheSumOfSquares(n int) int {
	var result int
	for ; n > 0 ; {
		var square int = (n % 10) * (n % 10)
		result += square
		n /= 10
	}

	return result
}

func isHappy(n int) bool {
	var AppereanceOfSums map[int]bool = map[int]bool{}
	for n != 1 {
		n = getTheSumOfSquares(n)
		if AppereanceOfSums[n] {
			return false
		} else {
			AppereanceOfSums[n] = true
		}
	}

	return true
}

File: LeetCode/cpp/isHappy.cxx
#include <unordered_map>

class Solution {
public:
    bool isHappy(int n) {
        std::unordered_map<int, bool> appereanceOfSums;

        while (n != 1)
        {
            n = getTheSumOfSquares(n);
            if (appereanceOfSums[n]) {
                return false;
            } else {
                appereanceOfSums[n] = true;
            }
        }

        return true;
    }

private:
    int getTheSumOfSquares (int n) {
        int result = 0;

        while (n > 0)
        {
            int square = n % 10;
            result += square * square;
            n /= 10;
        }

        return result;
    }
};